user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/BatchRegistrationDialog2.java

```diff
@@ -0,0 +1,547 @@
+      ComponentValueChangeEvent<TextField, String> batchNameChangedEvent) {
+    /* do nothing */
+  }
+
+  private void onRemoveLastRowClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.removeLastRow();
+  }
+
+  private void onAddRowClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.setValidationMode(ValidationMode.LAZY);
+    spreadsheet.addRow(new SampleInfo());
+    spreadsheet.setValidationMode(ValidationMode.EAGER);
+  }
+
+  private void onPrefillClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.setValidationMode(ValidationMode.LAZY);
+    spreadsheet.resetRows();
+    for (SampleInfo sampleInfo : prefilledSampleInfos()) {
+      spreadsheet.addRow(sampleInfo);
+    }
+    spreadsheet.setValidationMode(ValidationMode.EAGER);
+  }
+
+  @Override
+  protected void onConfirmClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.validate();
+    if (spreadsheet.isInvalid()) {
+      return;
+    }
+    if (batchNameField.isInvalid()) {
+      return;
+    }
+    fireEvent(new ConfirmEvent(this, clickEvent.isFromClient(),
+        new Data(batchNameField.getValue(), spreadsheet.getData())));
+  }
+
+  @Override
+  protected void onCancelClicked(ClickEvent<Button> clickEvent) {
+    fireEvent(new CancelEvent(this, clickEvent.isFromClient()));
+  }
+
+  public void addCancelListener(ComponentEventListener<CancelEvent> listener) {
+    addListener(CancelEvent.class, listener);
+  }
+
+  public void addConfirmListener(ComponentEventListener<ConfirmEvent> listener) {
+    addListener(ConfirmEvent.class, listener);
+  }
+
+  public static class CancelEvent extends ComponentEvent<BatchRegistrationDialog2> {
+
+    /**
+     * Creates a new event using the given source and indicator whether the event originated from
+     * the client side or the server side.
+     *
+     * @param source     the source component
+     * @param fromClient <code>true</code> if the event originated from the client
+     *                   side, <code>false</code> otherwise
+     */
+    public CancelEvent(BatchRegistrationDialog2 source, boolean fromClient) {
+      super(source, fromClient);
+    }
+  }
+
+  public static class ConfirmEvent extends ComponentEvent<BatchRegistrationDialog2> {
+
+    public record Data(String batchName, List<SampleInfo> samples) {
+
+    }
+
+    private final Data data;
+
+    /**
+     * Creates a new event using the given source and indicator whether the event originated from
+     * the client side or the server side.
+     *
+     * @param source     the source component
+     * @param fromClient <code>true</code> if the event originated from the client
+     *                   side, <code>false</code> otherwise
+     */
+    public ConfirmEvent(BatchRegistrationDialog2 source, boolean fromClient, Data data) {
+      super(source, fromClient);
+      this.data = data;
+    }
+
+    public Data getData() {
+      return data;
+    }
+  }
+
+  public static class SampleInfo {
```

Comment:
Major NIT: 
```suggestion
  public static class SampleEntry {
```

Reply:
As this is the information regarding a sample and the sample is not yet entered anywhere, I would prefer to keep SampleInfo